### PR TITLE
MULTIARCH-3965: Check if PER is enabled in the target PowerVS workspace

### PR DESCRIPTION
**Pull request by miyamotoh**
Power Edge Router (aka. "PER") needs to be enabled in the target region to connect PowerVS to VPC, and thus is required for IPI to proceed. Hence this new validation check before doing anything substantial with Terraform. 

Signed-off-by: Hiro Miyamoto <miyamotoh@us.ibm.com>

### CODE REVIEW

**Comment by mjturek** [comment]
Closes [MULTIARCH-3965 ](https://issues.redhat.com/browse/MULTIARCH-3965)

**Comment by hamzy** [comment]
FYI and not a review, but fix the golint errors

File: pkg/asset/installconfig/powervs/client.go [hunk 1]
```diff
@@ -746,3 +750,30 @@ func (c *Client) ServiceInstanceIDToCRN(ctx context.Context, id string) (string,
 
 	return "", nil
 }
+
+// GetDatacenterCapabilities retrieves the capabilities of the specified datacenter.
+func (c *Client) GetDatacenterCapabilities(ctx context.Context, region string) (map[string]bool, error) {
```
Comment (barbacbd) [inline pkg/asset/installconfig/powervs/client.go:755]: Can we wrap or format the errors here to include the original error ?

`fmt.Errorf("...: %w", err)`

Comment (miyamotoh) [reply pkg/asset/installconfig/powervs/client.go:755]: Definitely, good catch! Reflected in the latest force-push.

File: pkg/asset/installconfig/powervs/client.go [hunk 2]
```diff
@@ -746,3 +750,30 @@ func (c *Client) ServiceInstanceIDToCRN(ctx context.Context, id string) (string,
...
+	var err error
+	if c.BXCli.PISession == nil {
+		err = c.BXCli.NewPISession()
+		if err != nil {
+			return nil, err
```
[the remark this answers is not in the document]
Comment (miyamotoh) [reply pkg/asset/installconfig/powervs/client.go]: Addressed in the latest force-push. PTAL.

**Comment by miyamotoh** [comment]
@r4f4 @mjturek could you please review and approve?

File: pkg/asset/installconfig/platformprovisioncheck.go [hunk 3]
```diff
@@ -160,6 +160,11 @@ func (a *PlatformProvisionCheck) Generate(dependencies asset.Parents) error {
 			return err
 		}
 
+		err = powervsconfig.ValidatePERAvailability(client, ic.Config)
+		if err != nil {
+			return err
+		}
+
```
Comment (r4f4) [inline pkg/asset/installconfig/platformprovisioncheck.go:167]: Doesn't need to be addressed in this PR, but maybe `powervs` can follow the pattern of the other platforms and use a `ValidateForProvisioning` function that encapsulates all the different validations needed.

File: pkg/asset/installconfig/powervs/validation.go [hunk 4]
```diff
@@ -48,6 +48,20 @@ func validateMachinePool(fldPath *field.Path, machinePool *types.MachinePool) fi
...
+// ValidatePERAvailability ensures the target datacenter has PER enabled.
+func ValidatePERAvailability(client API, ic *types.InstallConfig) error {
+	capabilities, err := client.GetDatacenterCapabilities(context.TODO(), ic.PowerVS.Zone)
+	if err != nil {
+		return err
+	}
+	const per = "power-edge-router"
+	if !capabilities[per] {
+		return fmt.Errorf("%s is not available at: %s", per, ic.PowerVS.Zone)
+	}
```
Comment (r4f4) [inline pkg/asset/installconfig/powervs/validation.go:60]: Could `"power-edge-router"` not show up in the set of capabilities? If so, maybe it's better to separate the error types (i.e. "capability not found" vs "capability not available")?

File: pkg/asset/installconfig/powervs/validation_test.go [hunk 5]
```diff
@@ -113,6 +113,18 @@ var (
...
+	mapWithPERFalse  = map[string]bool{
+		"disaster-recover-site": true,
+		"power-edge-router":     false,
+		"vpn-connections":       true,
+	}
+	mapWithPERTrue = map[string]bool{
+		"disaster-recover-site": true,
+		"power-edge-router":     true,
+		"vpn-connections":       true,
+	}
```
Comment (r4f4) [inline pkg/asset/installconfig/powervs/validation_test.go:127]: For completeness, a test case where "power-edge-router" is not available.

**Comment by r4f4** [comment]
Jira issues are now required in all PRs.

File: pkg/asset/installconfig/powervs/validation_test.go (hunk 5 above)
Comment (miyamotoh) [reply pkg/asset/installconfig/powervs/validation_test.go:128]: Addressed in the latest force-push. PTAL.

File: pkg/asset/installconfig/powervs/validation.go (hunk 4 above)
Comment (miyamotoh) [reply pkg/asset/installconfig/powervs/validation.go]: Addressed in the latest force-push. PTAL.

**Review by r4f4 — COMMENTED** [review]
/approve

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by mjturek** [comment]
We cannot get `openstack-manifests` test to pass. Is there an outage with this test?

**Comment by r4f4** [comment]
> We cannot get `openstack-manifests` test to pass. Is there an outage with this test?

Yes. See the fix at https://github.com/openshift/installer/pull/7617

**Pull request merged**